src/search.ts

```diff
@@ -36,6 +37,12 @@ export interface SearchResult {
     lastResultUrl?: string;
 }
 
+export interface SearchFilter {

```

Comment:
After some further thought, I think this interface can be
```
export interface SearchFilter {
    key: string;
    value: string[];
    operator: '==' | '!=' | '>' | '<' | '>=' | '<=';
    logicalOperator: 'AND' | 'OR';
}
```

We want to make `logicalOperator` a required field so that if `value` has multiple strings, we will definitely know the relationship between those values. If `value` only has one string, `logicalOperator` doesn't really apply, so the value there can be either `AND` or `OR`. It doesn't matter.

The alternative is to make `logicalOperator` an optional field. This is dangerous, because this allows for the use case where `value` has multiple strings, but there is no `logicalOperator` defined.

Reply:
Agreed. Currently all the fields in this interface will be required. I did use `comparisonOperator` instead of the plain `operator` name to differentiate from the `logicalOperator` field. I also added another `comparisonOperator` value `~` to represent ES' `query_string` queries that perform looser searches.